ADAL/src/telemetry/ADTelemetry.m

```diff
@@ -67,6 +67,12 @@ - (void)registerDispatcher:(id<ADDispatcher>)dispatcher
     @synchronized(self)
     {
         SAFE_ARC_RELEASE(_dispatcher);
+        if (!dispatcher)
+        {
+            _dispatcher = nil;
+            return;
+        }
+        
```

Comment:
if user passes in nil as dispatcher, we simply set _dispatcher to nil, such that we won't end up collecting a bunch of events and has no where to dispatch and discard them.